docs/source/guides/scripting/customwavecreation.rst

```diff
@@ -0,0 +1,479 @@
+
+- ``Shop Position``
+- ``Dropship Position``
+- ``Dropship Angles``
+- ``int index = 1``
+- ``index=1``
+- ``array<WaveEvent>`` and  ``waveEvents.append``
+
+Here is a basic setup of the things mentioned above that would let the script run:
+
+.. code-block:: 
+
+    global function initFrontierDefenseData
+    void function initFrontierDefenseData()
+    {
+    shopPosition = < -3862,1267,1060>
+	FD_spawnPosition = < -838,2629,1592>
+	FD_spawnAngles = <0,180,0>
+
+    int index = 1
+    array<WaveEvent> wave1
+    wave1.append(CreateLegionTitanEvent(<4060,771,1100>, <0,180,0>,"",0))
+    waveEvents.append(wave1)
+    index=1
+    array<WaveEvent> wave2
+    wave2.append(CreateLegionTitanEvent(<4060,771,1100 >, <0.,180,0.>,"",0))
+    waveEvents.append(wave2)
+
+    }
+
+What this script does is on the first wave when possible, it will spawn a Legion Titan
+at the coordinates and angles when the script is able to for the first wave. 
+Afterwards, the second wave starts and spawns yet another Legion Titan at the 
+same coordiantes/angles when available to do so. But what does this all mean? Well, I'll
+be explaining below what everything means with some extra event types and variables.
+
+Explaining What Everything Does in the Script 
+---------------------------------------------
+
+Here is another example of a mission I made to better explain 
+some other Event types that you can use in your own script:
+
+.. code-block::
+
+    global function initFrontierDefenseData
+    void function initFrontierDefenseData()
+    {
+        shopPosition = < -3862,1267,1060 >
+        shopAngles =  <0,30,0>
+        FD_spawnPosition = < -838,2629,1592 >
+        FD_spawnAngles = < 0,180,0 >
+
+        int index = 1
+
+        array<WaveEvent> wave1
+        wave1.append(CreateSmokeEvent(<4060,771,1004>,90,index++))
+        wave1.append(CreateWaitForTimeEvent(5,index++))
+        wave1.append(CreateWarningEvent(FD_IncomingWarnings.ArcTitanAlt,index++))
+        wave1.append(CreateArcTitanEvent(<4300,771,1004 >,<0,180,0>,"",index++))
+        wave1.append(CreateDroppodGruntEvent(<3733,100,1004 >,"",0))
+        waveEvents.append(wave1)
+        index=1
+        array<WaveEvent> wave2
+        wave2.append(CreateWarningEvent(FD_IncomingWarnings.LightWave,index++))
+        wave2.append(CreateDroppodStalkerEvent(<2851,4146,1000>,"",index++))
+        wave2.append(CreateLegionTitanEvent(<4060,771,1004>,<0,180,0>,"",index++))
+        wave2.append(CreateWaitForTimeEvent(3,index++))
+        wave2.append(CreateWarningEvent(FD_IncomingWarnings.MortarTitanIntro,index++))
+        wave2.append(CreateMortarTitanEvent(<2176,-3541,878>,<0,180,0>,index++))
+        wave2.append(CreateWaitForTimeEvent(3,index++))
+        wave2.append(CreateMortarTitanEvent(<2240,4887,1011>,<0,180,0>,0))
+        waveEvents.append(wave2)
+
+    }
+
+Init Functions
```

Comment:
its just one function and use the ``.. cpp:function::`` like in the rest of the docs